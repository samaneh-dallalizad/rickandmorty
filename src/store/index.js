// Create a new store instance.
import {createStore} from "vuex";

export const store = createStore({
    state () {
        return {
            characters: []
        }
    },
    mutations: {
        setCharacters (state, characters) {
            state.characters = characters.slice(0,10)
        }
    },
    actions: {
        async getCharacters ({commit}) {
            let response = await fetch("https://rickandmortyapi.com/api/character")
            let data = await response.json()
            commit("setCharacters", data["results"])
        }
    }
})
