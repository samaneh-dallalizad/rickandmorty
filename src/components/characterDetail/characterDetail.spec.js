import { expect, it, describe } from 'vitest'
import Vue from 'vue';
import {mount} from '@vue/test-utils'
import CharacterDetailComponent from './characterDetail.vue';
import {store} from '../../store';
import {fetch} from 'whatwg-fetch'

describe('HomepageComponent', () => {
    it('CharacterList must always be the same', async () => {
        await store.dispatch('getCharacters')
        const wrapper = mount(CharacterDetailComponent, {
            propsData: {
                character: store.state.characters[0],
                active: true
            },
        })
        expect(wrapper.element).toMatchSnapshot()
    })
})