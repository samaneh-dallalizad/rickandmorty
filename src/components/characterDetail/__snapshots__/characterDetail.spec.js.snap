// Vitest Snapshot v1

exports[`HomepageComponent > CharacterList must always be the same 1`] = `
<div
  data-v-app=""
>
  
  <!--modal-->
  <div
    class="modal is-active"
  >
    <div
      class="modal-background"
    />
    <div
      class="modal-card"
    >
      <!--header-->
      <header
        class="modal-card-head"
      >
        <!--title/character_name-->
        <p
          class="modal-card-title"
        >
          character: Rick Sanchez
        </p>
        <!--close button-->
        <button
          aria-label="close"
          class="button is-delete is-danger delete"
        >
          Close 
        </button>
      </header>
      <!--body/content-->
      <section
        class="modal-card-body"
      >
        <div
          class="columns"
        >
          <!--character_image-->
          <div
            class="column is-4"
          >
            <figure
              class="image is-128x128 mx-auto mb-10px"
            >
              <img
                alt="{{ character?.origin.name }}"
                class="is-rounded"
                src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
              />
            </figure>
          </div>
          <div
            class="column is-8"
          >
            <ul>
              <!--character_Origin-->
              <li>
                <strong
                  class="is-inline has-text-primary"
                >
                   Origin: 
                </strong>
                 Earth (C-137)
              </li>
              <!--character_status-->
              <li>
                <strong
                  class="is-inline has-text-primary"
                >
                  Status: 
                </strong>
                 Alive
              </li>
              <!--character_gender-->
              <li>
                <strong
                  class="is-inline has-text-primary"
                >
                   gender: 
                </strong>
                 Male
              </li>
              <!--character_species-->
              <li>
                <strong
                  class="is-inline has-text-primary"
                >
                   species: 
                </strong>
                 Human
              </li>
              <!--character_lacation-->
              <li>
                <strong
                  class="is-inline has-text-primary"
                >
                   location: 
                </strong>
                 Citadel of Ricks
              </li>
            </ul>
          </div>
        </div>
      </section>
    </div>
  </div>
  
</div>
`;
